src/tribler/core/components/libtorrent/restapi/downloads_endpoint.py

```diff
@@ -50,7 +50,7 @@ def _safe_extended_peer_info(ext_peer_info):
         return ensure_unicode(ext_peer_info, "utf8")
     except UnicodeDecodeError:
         # We might have some special unicode characters in here
-        return ''.join([chr(c) for c in ext_peer_info])
+        return ''.join(map(chr, ext_peer_info))
```

Comment:
It's not your fault that there aren't any tests for this line.

Feel free to add the following:

```python
@patch('tribler.core.components.libtorrent.restapi.downloads_endpoint.ensure_unicode',
       Mock(side_effect=UnicodeDecodeError('', b'', 0, 0, '')))
def test_safe_extended_peer_info():
    # Test that we return the string mapped by `chr` in the case of `UnicodeDecodeError`
    extended_peer_info = _safe_extended_peer_info(b'abcd')
    assert extended_peer_info == 'abcd'
```

Reply:
Thanks!